00_Base/src/model/DTO/admin/AdminLocationDTO.ts

```diff
@@ -0,0 +1,153 @@
+import { GeoLocation } from '../../GeoLocation';
+import {
+  IsArray,
+  IsBoolean,
+  IsNotEmpty,
+  IsNumber,
+  IsObject,
+  IsString,
+  MaxLength,
+  ValidateNested,
+} from 'class-validator';
+import { Optional } from '../../../util/decorators/optional';
+import { Type } from 'class-transformer';
+
+const LOCATION_INVALID_MSG =
+  'New locations must have party_id, country_code, name, address, city, state, postal_code, and country.';
+const EVSE_INVALID_MSG = 'EVSEs must have id and station_id.';
+const CONNECTOR_INVALID_MSG = 'Connectors must have an id.';
+
+// TODO add remaining OCPI-specific properties
+export class AdminLocationDTO {
+  // id is the Citrine Location database id, not the OcpiLocation's database id
+  @IsNumber()
+  @Optional()
+  id?: number;
+
+  @IsString()
+  @MaxLength(2)
+  @Optional()
+  country_code!: string;
+
+  @IsString()
+  @MaxLength(3)
+  @Optional()
+  party_id!: string;
+
+  @IsString()
+  @Optional()
+  name?: string;
+
+  @IsString()
+  @Optional()
+  address?: string;
+
+  @IsString()
+  @Optional()
+  city?: string;
+
+  @IsString()
+  @Optional()
+  state?: string;
+
+  @IsString()
+  @Optional()
+  postal_code?: string;
+
+  @IsString()
+  @Optional()
+  country?: string;
+
+  @IsObject()
+  @Optional()
+  @Type(() => GeoLocation)
+  @ValidateNested()
+  coordinates?: GeoLocation;
+
+  @IsString()
+  @Optional()
+  time_zone?: string;
+
+  @IsBoolean()
+  @Optional()
+  publish?: boolean;
+
+  @IsArray()
+  @Optional()
+  evses?: AdminEVSEDTO[];
+
+  static IS_LOCATION_INVALID = (
```

Comment:
Shouldn't the validation annotations above take care of validation?

Why is this method needed?